### ISSUE TITLE
Allow configuring classifier keywords

### ISSUE THREAD

**Issue by nodiscc** [bitmagnet-io/bitmagnet#182]
- [x] I have checked the existing issues to avoid duplicates
- [x] I have redacted any info hashes and content metadata from any logs or screenshots attached to this issue

### Is your feature request related to a problem? Please describe

As the number of indexed torrents grows, so does the number of unclassified torrents. Torrents in the `Unknown` category amount to ~65% of the total on my current testing instance. On the long term, centralizing all keywords in https://github.com/bitmagnet-io/bitmagnet/blob/main/internal/classifier/keywords/words.go and having to maintain a fork or submitting a PR here to add the keywords and wait for a new release before running a reprocess/reclassify, will become impractical.

### Describe the solution you'd like

The ability to configure custom classifier keywords as an admin of a bitmagnet instance, in the configuration file.

Example:

```yaml
classifier:
  keywords:
    music:
      - 'FLAC'
      - '320 kbps'
    game:
      - 'itch.io'
      - '-RELEASEGROUPNAME'
    shows:
      - '\.S[0-0][0-9]E[0-9][0-9]\.' # regex support?
    movie:
    ...
```

### Describe alternatives you've considered

- Submitting my own classifier keywords here (does not scale, also subjective with possible false positives)
- Forking and maintaining keywords in a separate fork (impractical compared to the config file-based solution)

### Additional

**Comment by mgdigital** [comment]
Hey, this is a great idea, especially as there are lots of probably useful keywords that I don't want to add in the code here.

I'd go further and say that the whole classifier should be declarative in YAML format like this, not just these keywords that are 1 part of 1 of the classifiers.

> regex support?

Maybe - my reservation about this is I've used https://github.com/hedhyw/rex to create some truly gnarly regexes (e.g. see https://github.com/bitmagnet-io/bitmagnet/blob/main/internal/classifier/video/parse.go) - which could be output in raw form here but they'd be difficult to read or update.

I've been talking with @davis about a sort-of related thing whereby the CLI could output a template `config.yaml`, which this could be part of.

**Comment by leofidus** [comment]
Would it be a viable solution to keep the complex logic in the parsers, and merely provide an interface for it in the classifier config? Maybe with special keywords like `contains:filetype:mp4` and `has:movie:title`, `has:movie:year`.

The biggest complexity (from an UX perspective) might be expressing which conditions have to occur together and which are alternatives. One idea might be to change the way the parser works completely and instead do it via a voting system: each condition can add and subtract points to specific categories, and the category with the most points wins out (if it clears some threshold).

For example
```
classifier:
  - '\.S[0-0][0-9]E[0-9][0-9]\.':
    shows: +10
  - contains:filetype:mp4:
    shows: +2
    movies: +2
  - contains:filetype:mp3:
    music: +4
  - 'FLAC':
    music: +10
  - \([0-9]{4}\):
    movies: +2
```

That would be a pretty big change though, and probably require significant fine tuning. It would however offer a lot of control in a fairly simple yaml format, and be straightforward to extend.

**Comment by davispuh** [comment]
+1 for configurable classifier but I disagree with putting it in `config.yaml`

I think to support all use cases best would be supporting classifier plugins and providing default regexp classifier plugin that reads from `keywords.yaml`. So most people could simply update that but some who might want to do more advanced stuff could write their own plugin that uses their own private DBs and whatnot.

**Comment by nodiscc** [comment]
I don't think too much thought should go into this. Configuration through `config.yaml` would work for the vast majority, if not all use cases. A plugin architecture might be nice to have later later, but this simple feature should not be blocked by theoretical "what if some people want to..."
